cmake/HPX_AddConfigTest.cmake

```diff
@@ -511,6 +512,14 @@ macro(hpx_check_for_cxx11_std_unordered_set)
 endmacro()
 
 ###############################################################################
+macro(hpx_check_for_cxx14_auto)
+  add_hpx_config_test(HPX_WITH_CXX14_AUTO
+    SOURCE cmake/tests/cxx14_auto.cpp
+    FILE ${ARGN}
+    CMAKECXXFEATURE cxx14_auto_type)
```

Comment:
There is no C++ feature detection property in cmake called `cxx14_auto_type` (please see https://cmake.org/cmake/help/v3.9/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html)